docker/server/Dockerfile.ubuntu

```diff
@@ -40,82 +45,78 @@ ARG DIRECT_DOWNLOAD_URLS=""
+        wget --progress=bar:force:noscroll "${single_binary_location_url}" -O /tmp/clickhouse/clickhouse; \
+        chmod +x /tmp/clickhouse/clickhouse; \
+        /tmp/clickhouse/clickhouse install --user "clickhouse" --group "clickhouse"; \
+    fi; \
+    \
+    # A fallback to installation from ClickHouse repository
+    if ! clickhouse local -q "SELECT ''" > /dev/null 2>&1; then \
+        apt-get update; \
+        apt-get install --yes --no-install-recommends \
             apt-transport-https \
             ca-certificates \
             dirmngr \
             gnupg2 \
-        && mkdir -p /etc/apt/sources.list.d \
-        && GNUPGHOME=$(mktemp -d) \
-        && GNUPGHOME="$GNUPGHOME" gpg --no-default-keyring \
+            ;\
+        mkdir -p /etc/apt/sources.list.d; \
+        GNUPGHOME=$(mktemp -d); \
+        GNUPGHOME="$GNUPGHOME" gpg --no-default-keyring \
             --keyring /usr/share/keyrings/clickhouse-keyring.gpg \
-            --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8919F6BD2B48D754 \
-        && rm -rf "$GNUPGHOME" \
-        && chmod +r /usr/share/keyrings/clickhouse-keyring.gpg \
-        && echo "${REPOSITORY}" > /etc/apt/sources.list.d/clickhouse.list \
-        && echo "installing from repository: ${REPOSITORY}" \
-        && apt-get update \
-        && for package in ${PACKAGES}; do \
-            packages="${packages} ${package}=${VERSION}" \
-        ; done \
-        && apt-get install --allow-unauthenticated --yes --no-install-recommends ${packages} || exit 1 \
-        && rm -rf \
+            --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8919F6BD2B48D754; \
+        rm -rf "$GNUPGHOME"; \
+        chmod +r /usr/share/keyrings/clickhouse-keyring.gpg; \
+        echo "${REPOSITORY}" > /etc/apt/sources.list.d/clickhouse.list; \
+        echo "installing from repository: ${REPOSITORY}"; \
+        apt-get update; \
+        for package in ${PACKAGES}; do \
+            packages="${packages} ${package}=${VERSION}"; \
+        done; \
+        apt-get install --allow-unauthenticated --yes --no-install-recommends ${packages} || exit 1; \
+        rm -rf \
             /var/lib/apt/lists/* \
-            /var/cache/debconf \
-            /tmp/* \
-        && apt-get autoremove --purge -yq libksba8 \
-        && apt-get autoremove -yq \
-    ; fi
+            /var/cache/debconf; \
+        apt-get autoremove --purge -yq libksba8; \
+        apt-get autoremove -yq; \
+    fi; \
+    \
+    rm -rf /tmp/*;
 
 # post install
 # we need to allow "others" access to clickhouse folder, because docker container
 # can be started with arbitrary uid (openshift usecase)
-RUN clickhouse-local -q 'SELECT * FROM system.build_options' \
-    && mkdir -p /var/lib/clickhouse /var/log/clickhouse-server /etc/clickhouse-server /etc/clickhouse-client \
-    && chmod ugo+Xrw -R /var/lib/clickhouse /var/log/clickhouse-server /etc/clickhouse-server /etc/clickhouse-client
+RUN set -ex; \
```

Comment:
Also didn't understand.

Reply:
This makes the first error stops the build which helps understand build errors a lot easier. Otherwise they get hidden or burried in logs.